supbot/routers/error_handling_tgm.py

```diff
@@ -0,0 +1,13 @@
+import logging
+from typing import Literal
+
+from aiogram import Router
+from aiogram.types import ErrorEvent
+
+router = Router()
+
+
+@router.errors()
+async def error_handler(error: ErrorEvent) -> Literal[True]:
+    logging.error(f"An error has occurred: {error.exception}", exc_info=True)
```

Comment:
question: why `exc_info=True` вместо чего-то вроде `exec_info=error.exception`

Reply:
issue: ошибка недескриптивная. То, что произошла ошибка мы поняли по уровню лога. Где-когда-что-зачем — непонятно. Как минимум (а может так и достаточно) написать, что это в боте

Reply:
suggestion: Закинуть побольше инфы из `ErrorEvent` в `extra=`, интеграция с сентри это будет очень красиво ловить

Reply:
> question: why `exc_info=True` вместо чего-то вроде `exec_info=error.exception`

exc_info=True добавит traceback к сообщению
если exec_info имеется ввиду exc_info там ведь только True или False, а error.exception это сообщение об ошибке


> issue: ошибка недескриптивная. То, что произошла ошибка мы поняли по уровню лога. Где-когда-что-зачем — непонятно. Как минимум (а может так и достаточно) написать, что это в боте

Написал, что это в телеграм боте. Подробнее будет в трейсбэке. Если что-то дополнительно нужно можно добавить в extra

Reply:
> exc_info там ведь только True или False

Не угадал, из [доки python](https://docs.python.org/3/library/logging.html#logging.Logger.debug)

> If exc_info does not evaluate as false, it causes exception information to be added to the logging message. If an exception tuple (in the format returned by sys.exc_info()) or **an exception instance is provided, it is used**; otherwise, sys.exc_info() is called to get the exception information.

Reply:
issue закрыто, question остался

Reply:
> > exc_info там ведь только True или False
> 
> Не угадал, из [доки python](https://docs.python.org/3/library/logging.html#logging.Logger.debug):
> 
> > If exc_info does not evaluate as false, it causes exception information to be added to the logging message. If an exception tuple (in the format returned by sys.exc_info()) or **an exception instance is provided, it is used**; otherwise, sys.exc_info() is called to get the exception information.

не знал. Сделал exc_info=sys.exc_info() чтобы было явно, как по доке

Reply:
Ты вставил дефолт, так не делают. В доке прямым тестом сказано передать инстанс exception-а (e)

Reply:
оставил инстанс exception-а